src/globus_sdk/experimental/globus_app/app.py

```diff
@@ -227,16 +227,75 @@ def _initialize_authorizer_factory(self) -> None:
         authorize requests.
         """
 
+    def login(
+        self,
+        *,
+        auth_params: GlobusAuthorizationParameters | None = None,
+        force: bool = False,
+    ) -> None:
+        """
+        Log an auth entity into the app, if needed, storing the resulting tokens.
+
+        A login flow will be performed if:
+            * ``auth_params`` are provided,
+            * ``force`` is set to True, or
+            * ``self.login_required()`` evaluates to True.
+
+        :param auth_params: An optional set of authorization parameters to establish
+            requirements and controls for the login flow.
+        :param force: If True, perform a login flow even if one is not necessary.
```

Comment:
I was under the impression that `.login_required()` might not perfectly know if login is required (I'm thinking of 30-minute timeouts), so it might be helpful to rephrase this sentence.

```suggestion
        :param force: If True, perform a login flow even if one doesn't appear to be necessary.
```

Reply:
Sure, I can update